Create Intake

erinLeee opened the pull request:
Created Intake subsystem and IntakeCargo command

potttaaatttooo committed:
Create Intake

Created Intake subsystem and IntakeCargo command

jyue86 commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -4,14 +4,30 @@
+   CANSparkMax motorController2; 
+
+  public Intake() {
+    motorController1 = new CANSparkMax(0, MotorType.kBrushless);
```
Remember to change the IDs of the motor controllers since they're being used by the drivetrain.

rice31415 replied:
the id should also be a constant

rice31415 replied:
it might also be better to rename the variables to something like topSpark and bottomSpark so that its more clear what it corresponds to on the robot

jyue86 commented on vendordeps/REVLib.json:
```diff
@@ -0,0 +1,73 @@
+{
```
@rice31415 Do you think that adding the vendor libraries may cause a git conflict since the other branches probably have this committed change as well?

rice31415 replied:
Nothing else has used a Spark MAX yet so I don't think the other branches actually have the REV dependency yet, but the climber might also have this change once that gets worked on

jyue86 commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -21,10 +24,31 @@
   private final ExampleCommand m_autoCommand = new ExampleCommand(m_exampleSubsystem);
+  LogitechGamingPad drivePad;
+
+  DriveTrain driveTrain;
+  Shooter shooter;
+  Intake intake;
```
```suggestion
  Intake intake;
```
Since this branch deals solely w/ the Intake subsystem, it is okay to remove the other 2 subsystems. The branches that focus on the drivetrain and shooter will declare and initialize their respective subsystems.

jyue86 commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -21,10 +24,31 @@
+    
+    drivePad = new LogitechGamingPad(0);
+
+    driveTrain = new DriveTrain();
+    shooter = new Shooter();
+    intake = new Intake();
```
```suggestion
    intake = new Intake();
```
Similar reasoning as to previous comment dealing w/ the initialization of the subsystems.

jyue86 reviewed (changes requested):
Minor changes needed but looking good!

rice31415 commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -4,14 +4,30 @@
+  }
+
+  public void intakeBalls(double speed){
+    motorController1.set(speed);
+    motorController2.set(-speed);
```
Because of how the intake is going to work, you'll need to be able to control the direction of each individual motor controller, so it should probably be two separate methods, one for each motor

You'll also need two different commands, one for intaking the cargo which you technically have already, but also another command to feed the ball to the shooter. Ask Nathan about which motor spins which direction for each command. 

Also create a stop method for the intake

potttaaatttooo committed:
Update intake methods

Fixed the Constants, created methods for each motor, and created the Stop() method

potttaaatttooo committed:
Intake and IntakeCargo update

updated code for Intake and IntakeCargo, was able to add boolean for isIntaking.

Eventide-03 committed:
Set x and right bumper

x intakes the game piece to the shooter, while right bumper turns on the intake

Eventide-03 committed:
Changed indentation and removed wildcards

rice31415 reviewed (approved):
Should probably remove unused imports and add comments but fine for now so they can work on shooting command sequence

jyue86 commented on src/main/java/frc/robot/commands/IntakeCargo.java:
```diff
@@ -0,0 +1,49 @@
+
+  // Called when the command is initially scheduled.
+  @Override
+  
```
```suggestion
```
Remove random empty line between the @override annotation and the function signature.

Commit:
Merge branch 'main' into elee_aluo/Intake
